src/app/clusters/wifi-network-management-server/wifi-network-management-server.cpp

```diff
@@ -0,0 +1,142 @@
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "wifi-network-management-server.h"
+
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/reporting.h>
+#include <app/util/config.h>
+#include <lib/core/Global.h>
+#include <lib/support/CodeUtils.h>
+
+#include <functional>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::WiFiNetworkManagement::Attributes;
+using namespace std::placeholders;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WiFiNetworkManagement {
+
+Global<Server> gServerInstance;
+
+Server & Server::Instance()
+{
+    return gServerInstance.get();
+}
+
+Server::Server() : AttributeAccessInterface(NullOptional, Id), CommandHandlerInterface(NullOptional, Id) {}
+
+CHIP_ERROR Server::Init(EndpointId endpoint)
+{
+    VerifyOrReturnError(endpoint != kInvalidEndpointId, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mEndpointId == kInvalidEndpointId, CHIP_ERROR_INCORRECT_STATE);
+
+    mEndpointId = endpoint;
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INTERNAL);
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Server::ClearNetworkCredentials()
+{
+    VerifyOrReturnError(!SsidSpan().empty() || !PassphraseSpan().empty(), CHIP_NO_ERROR);
+
+    mSsidLen = 0;
+    mPassphrase.SetLength(0);
+    MatterReportingAttributeChangeCallback(mEndpointId, Id, Ssid::Id);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR Server::SetNetworkCredentials(ByteSpan ssid, ByteSpan passphrase)
+{
+    VerifyOrReturnError(1 <= ssid.size() && ssid.size() <= sizeof(mSsid), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(1 <= passphrase.size() && passphrase.size() <= mPassphrase.Capacity(), CHIP_ERROR_INVALID_ARGUMENT);
+
+    VerifyOrReturnError(!SsidSpan().data_equal(ssid) || !PassphraseSpan().data_equal(passphrase), CHIP_NO_ERROR);
+
+    memcpy(mSsid, ssid.data(), ssid.size());
+    mSsidLen = static_cast<decltype(mSsidLen)>(ssid.size());
+
+    ReturnErrorOnFailure(mPassphrase.SetLength(passphrase.size()));
+    memcpy(mPassphrase.Bytes(), passphrase.data(), passphrase.size());
+
+    MatterReportingAttributeChangeCallback(mEndpointId, Id, Ssid::Id); // report SSID change even if only passphrase changed
```

Comment:
Why?  That's a very strange thing to do....  Please explain why that's happening.

Reply:
The spec has no event defined currently for reporting a change to the passphrase, so this was intended to be something of a placeholder until we tidy up the spec, but maybe it's better to just take it out. I've included this issue in https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/9234.

Reply:
Sounds good, thanks!